### PR TITLE
Explain how to go around rust macro limitations

### LINKED ISSUE #966: Rust macro has no access to parameters passed into `{% macro f(param) %}`
Not sure if this is a question, feature request or a bug.

From the documentation it's not clear to me in what scope rust macros are evaluated.
For instance, I have this Askama macro, which calls a Rust macro internally:

```jinja
{% macro stop(entity) %}
    {{ create_action!(entity, LightOn, {brightness:Some(0.1), color_temperature: None})|json }}
{% endmacro %}
```

In some situations this works perfectly, while in other situations this does not and results in a "cannot find value `entity` in this scope". (`entity` being the argument passed to the Askama macro.)

It does work for one situation where this macro is called in another template within a loop like this: `{% for entity in entities.values() %}`. In any case, I've the impression that it doesn't pick up the argument passed to the Askama macro.

Is that understanding correct? Is there something we can improve here to make this possible?

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #966.

Does it look good to you @jonathanslenders?

### CODE REVIEW

**Comment by jonathanslenders** [comment]
Perfect! Thanks for the quick response!

**Review by Kijewski — APPROVED** [review]
:+1:

File: book/src/template_syntax.md [hunk 1]
```diff
@@ -642,3 +642,38 @@ In here it's invalid because `arg2` is the second argument and would be used by
...
+## Calling Rust macros
+
+It is possible to call rust macros directly in your templates:
+
+```jinja
+{% let s = format!("{}", 12) %}
+```
+
+One important thing to note is that contrary to the rest of the expressions,
+askama cannot know if a token given to a macro is a variable or something
```
Comment (djc) [inline book/src/template_syntax.md:655]: Nit: I think we capitalize Askama elsewhere.

**Review by djc — APPROVED** [review]
Thanks!

**Pull request merged**